pkg/node/fsm/ng_state.go

```diff
@@ -296,6 +296,23 @@ func (a *NGState) mineMicro(
 		return a, nil, a.Errorf(errors.Wrap(err, "failed to broadcast generated microblock"))
 	}
 
+	blockchainHeight, err := a.baseInfo.storage.Height()
+	if err != nil {
+		return a, nil, a.Errorf(errors.Wrap(err, "failed to get blockchain height"))
+	}
+	// here the blockchainHeight is equal to lastBlockHeight because we are appending a microblock to the last block
+	lastBlockHeight := blockchainHeight
+	ok, err := a.baseInfo.storage.IsActiveLightNodeNewBlocksFields(lastBlockHeight)
+	if err != nil {
+		return a, nil, a.Errorf(err)
+	}
+	if ok {
+		sh, errSh := a.baseInfo.storage.SnapshotStateHashAtHeight(lastBlockHeight)
+		if errSh != nil {
+			return a, nil, a.Errorf(errSh)
+		}
+		micro.StateHash = &sh
+	}
```

Comment:
The result block will be invalid because it's signed and generated in `MicroMiner.Micro` method. `StateHash` should be calculated there and the block should be signed when this field is set.

Maybe modify `CreateBlock` constructor and add new fields to the its arguments?